VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+# Versioning
+
+This document describes the versioning policy for this repository. This policy
+is designed so the following goals can be achieved.
+
+**Users are provided a codebase of value that is stable and secure.**
+
+## Policy
+
+* Versioning of this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver
+          2.0](https://semver.org/spec/v2.0.0.html) with the following exceptions.
+            * New methods may be added to exported API interfaces. All exported
+              interfaces that fall within this exception will include the following
+              paragraph in their public documentation.
+
+              > Warning: methods may be added to this interface in minor releases.
+
+        * If a module is version `v2` or higher, the major version of the module
+          must be included as a `/vN` at the end of the module paths used in
+          `go.mod` files (e.g., `module go.opentelemetry.io/collector/v2`, `require
+          go.opentelemetry.io/collector/v2 v2.0.1`) and in the package import path
+          (e.g., `import "go.opentelemetry.io/collector/v2/component"`). This includes the
+          paths used in `go get` commands (e.g., `go get
+          go.opentelemetry.io/collector/v2@v2.0.1`.  Note there is both a `/v2` and a
+          `@v2.0.1` in that example. One way to think about it is that the module
+          name now includes the `/v2`, so include `/v2` whenever you are using the
+          module name).
+        * If a module is version `v0` or `v1`, do not include the major version in
+          either the module path or the import path.
+    * Modules will be used to encapsulate signals and components.
```

Comment:
Does this mean that each component will have (at least) a separate module for traces, metrics and logs? Are breaking changes acceptable on parts of a component related to experimental signals? What about other more overarching yet currently unstable parts of OpenTelemetry like semantic conventions?

Reply:
I think the decision was to not do this split, we have 2 options here:
1. Keep semver 0 and add a note that some APIs are stable (this is a stronger guarantee than semver has for v0).
2. Move to semver 1 and add a note that some APIs are unstable (this is a weaker guarantee than smever has for v1).

I personally prefer the first option because we will be semver compatible. When all signals are declared stabled then we can move the packages to semver v1.

/cc @punya for the discussion we had about not splitting that granular the collector code.

Reply:
Unfortunately I don't remember the details of why we decided not to split packages by signal (it may have been as simple as a desire to keep the already-large number of modules under control).

Regardless, I agree with @bogdandrutu's proposed approach of sticking to v0.x and documenting which APIs are stable. I believe this is a good trade-off because

* it respects SemVer (i.e. encourages consumers to think about the dependency they are taking rather than making a promise we'll break later)
* it's appropriate for the collector, which is the core of an extensible application, rather than a general purpose library

Reply:
Thanks for your answers :) I agree that option (1) mentioned by Bogdan above makes the most sense

Reply:
Option 1 of not breaking semver and adding clear labels to identify APIs that are stable works for me.

Reply:
@bogdandrutu do you believe it necessary to state in this document that APIs we expect to be stable at v0 should be noted, or is that a practice we can follow without stating it in the policy?